src/double_linked_list.py

```diff
@@ -35,7 +32,7 @@ def push(self, val):
         self._counter += 1
```

Comment:
if self.head is always assigned to be new head regardless, take it out of the conditional instead of writing it twice